strawberry/field.py

```diff
@@ -33,6 +35,29 @@ def __init__(self, field_definition: FieldDefinition):
             metadata=None,
         )
 
+    def __call__(self, resolver: Callable) -> Callable:
+        """Migrate the field definition to the resolver"""
+
+        # TODO: This does not copy the data and now two objects share the same
+        #       FieldDefinition. Presumably this field will be GC-ed soon
+        #       though?
+        field_definition = self._field_definition
+        # Note: field_definition.name is finalized in type_resolver._get_fields
+
+        resolver_name = to_camel_case(resolver.__name__)
+
+        field_definition.origin_name = resolver_name
+
+        field_definition.origin = resolver
+        field_definition.base_resolver = resolver
+        field_definition.arguments = get_arguments_from_resolver(resolver)
+
+        field_definition.type = get_return_annotation(field_definition)
+
+        resolver._field_definition = field_definition  # type: ignore
```

Comment:
This breaks when `resolver` is a classmethod. Apparently they are immutable? Not sure how to address...